app.js

```diff
@@ -0,0 +1,168 @@
+const RickAndMortyAPI = 'https://rickandmortyapi.com/api/character/';
+let favoriteCharactersData = [];
+let characterGridData = [];
+let currentPage = 1;
+
+// ARM BUTTONS
+const searchInput = document.getElementById('filter-search');
+const favoriteBar = document.getElementById('favorite-container');
+const cardContainer = document.getElementById('card-container');
+const nextPage = document.getElementById('next-page');
+const prevPage = document.getElementById('previous-page');
+searchInput.addEventListener('search', search);
+nextPage.addEventListener('click', page);
+prevPage.addEventListener('click', page);
+
+load();
+
+// This function is used to boot the app. 
+// It renders the favorites and the card grid.
+async function load() {
+    if (localStorage.getItem('favorites')) {
+        await getData(localStorage.getItem('favorites'), 'faves');
+        await showData('faves');
+    }
+    await getData(`?page=${currentPage}`, 'grid');
+    await showData('grid');
```

Comment:
One improvement that could be made here is to set constants for `'faves'` and `'grid'`. 

We use these strings throughout the app, and we want them to be the same every time. For convenience, we could give each of them a `const` to make them easier to work with (code editor would autocomplete them, and we would never have to worry about spelling errors).